ksql-functional-tests/src/test/resources/query-validation-tests/cube.json

```diff
@@ -0,0 +1,42 @@
+{
+  "comments": [
+    "Tests for the CUBE table function"
+  ],
+  "tests": [
+    {
+      "name": "cube two int columns",
+      "statements": [
+        "CREATE STREAM TEST (col1 INT, col2 INT) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT cube(as_array(col1, col2)) VAL FROM TEST;"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "key": 0, "value": {"col1": 1, "col2": 2}},
+        {"topic": "test_topic", "key": 1, "value": {"col1": 1, "col2": null}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": "0", "value": {"VAL": [1, 2]}},
+        {"topic": "OUTPUT", "key": "0", "value": {"VAL": [1, null]}},
+        {"topic": "OUTPUT", "key": "0", "value": {"VAL": [null, 2]}},
+        {"topic": "OUTPUT", "key": "0", "value": {"VAL": [null, null]}},
+        {"topic": "OUTPUT", "key": "1", "value": {"VAL": [1, null]}},
+        {"topic": "OUTPUT", "key": "1", "value": {"VAL": [null, null]}}
+      ]
+    },
+    {
+      "name": "cube two columns with udf on third",
```

Comment:
What you think about one case with a UDAF/group-by as well, closer to the query we would need for metric aggregation?